docs: improve air-gapped Kommander installation documentation (#3234)
mesosphere/dcos-docs-site: pages/ksphere/kommander/1.2/install-airgapped/index.md

  - management cluster must connect to the attached cluster's API server
  - management cluster must connect to load balancers created by some addons. For example, Thanos, part of the Prometheus addon, connects to those load balancers.

> - all the prerequisites in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#before-you-begin] in case of Konvoy clusters. For attached 3rd party clusters refer to the vendor's documentation.

Reviewer: Why do we want to remove this?

Author: IMO this doesn't have any value because what could a 3rd party vendor have to say about the prerequisites of attaching its cluster to Kommander? I guess nothing.

Reviewer: For me, it is more of a statement of delimiting responsibility: "Hey - we can tell you what Konvoy needs, but 3rd party clusters are not our responsibility and you need to take care of it yourself.

Author: But how should they take care of it? There's no way for us to provide any action specific to satisfying prerequisites because we don't know what the prerequisites are for non-Konvoy clusters, right? So what we could say is "hey, we cannot tell you the prerequisites for your 3rd-party cluster and you may just have to find them out yourself by trial and error" which IMHO is not a very positive messaging to confront the customer with.

Reviewer: OK, I see your point.

- all the prerequisites in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#before-you-begin] in case of Konvoy clusters.

## Control plane nodes

> Control plane nodes of Konvoy clusters should meet the minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#control-plane-nodes]. For 3rd-party clusters control plane requirements refer to the vendor specification.

Reviewer: Ditto

Author: same reasoning as above

  - management cluster must connect to load balancers created by some addons. For example, Thanos, part of the Prometheus addon, connects to those load balancers.

- all the prerequisites in [air-gapped Konvoy installation](https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#before-you-begin) in case of Konvoy clusters.

## Control plane nodes

Control plane nodes of Konvoy clusters should meet the minimal requirements outlined in [air-gapped Konvoy installation](https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#control-plane-nodes).

## Worker nodes

Worker nodes must meet the minimal requirements outlined in [air-gapped Konvoy installation](https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#worker-nodes).

## Operating system and services for all nodes

All nodes must meet the same minimal requirements outlined in [air-gapped Konvoy installation](https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#operating-system-and-services-for-all-nodes).

## Define the inventory file

Installing air-gapped Kommander does not require any changes in the `inventory.yaml` file.

# Configure the Kubernetes cluster

> The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires extra configuration. Make sure the `cluster.yaml` has all the changes outlined in [air-gapped Konvoy installation](https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry) documentation. On top of that, you need to edit your `cluster.yaml` as outlined below to satisfy the following requirements:

Reviewer 2: ```suggestion
The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires extra configuration. Make sure the `cluster.yaml` has all the changes outlined in [air-gapped Konvoy installation](https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry) documentation. On top of that, you need to edit your `cluster.yaml` as outlined below to meet the following requirements:
```

1.  Make sure Kommander can use the self-hosted charts repository running on top of the Konvoy cluster. It can not connect to the default one through the public Internet.
1.  Make sure Kommander can find and access the private Docker registry. The `registry_ip` variable in the code snippet below references the IP address of the available private Docker registry. You can omit the username and password lines if your registry does not require authentication.
1.  Reconfigure the Kommander controller to work in an air-gapped environment.

> This is what your `cluster.yaml` should eventually look like regarding the Kommander Addon configuration:

Reviewer 2: ```suggestion
Your `cluster.yaml` file should look similar to the following for Kommander Addon configuration:
```

# Configure the Kubernetes cluster

> The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires additional configuration. Ensure the `cluster.yaml` contains all the changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation. Follow this procedure:

Reviewer: What is the difference between extra and additional?

CC: @russurquhart1

**TODO**: How is the reader supposed to conduct these following 3 steps?

1.  Ensure Kommander can use the self-hosted charts repository running on top of the Konvoy cluster. It can not connect to the default one through the public Internet.

> 1.  Ensure  Kommander can find and access the private Docker registry. The `bastion_ip` variable in the code snippet below is the IP address of the available private Docker registry. You can omit the username and password lines if your registry does not require authentication.

Reviewer: We need to replace all the occurences of `bastion_ip` in the test and replace it with `registry_ip`. Otherwise it will be super confusing for our users. The fact that we are using bastion as registry host is just our way of doing things, users can be using totally different approach.

Author: Sounds good. Will do that.

> The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires additional configuration. Ensure the `cluster.yaml` contains all the changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation. Follow this procedure:

Reviewer 2: ```suggestion
The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires extra configuration. Ensure the `cluster.yaml` has all the changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation. Follow this procedure:
```

> **TODO**: How is the reader supposed to conduct these following 3 steps?

Author: @vespian any hints?

Reviewer: I believe that the line above already explains it:

`Ensure the `cluster.yaml` has all the changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation. Follow this procedure:`

What exactly is the difficulty here?

Author: The difficulty is to understand how the customer is supposed to conduct each of the steps. The reader sees a list of actions like "Ensure Kommander can use the self-hosted charts repository" but nothing tells him **how** to do that.

Reviewer: >but nothing tells him how to do that.

We are describing here changes in cluster.yaml. Should we tell our customers how to edit a text file? Maybe we can simply be more clear that all these steps are done by changing the cluster.yaml?

Author: Ok, so all 3 checks here are satisfied by the `cluster.yaml` edit outlined below?

Reviewer: Yes!

> **TODO**: How is the reader supposed to conduct these following 3 steps?

Reviewer 2: Should this be handed over to the next dev?

Author: I don't think so.

1.  Ensure Kommander can use the self-hosted charts repository running on top of the Konvoy cluster. It can not connect to the default one through the public Internet.

> 1.  Ensure  Kommander can find and access the private Docker registry. The `bastion_ip` variable in the code snippet below is the IP address of the available private Docker registry. Leave the lines specifying username and password out if your registry does not require authentication.

Reviewer 2: ```suggestion
1.  Ensure  Kommander can find and access the private Docker registry. The `bastion_ip` variable in the code snippet below is the IP address of the available private Docker registry. You can omit the username and password lines if your registry does not require authentication.
```

1.  Reconfigure the Kommander controller to work in an air-gapped environment.

> Update the Kommander addon configuration like the following:

Reviewer: This whole section is about reconfiguring  an addon, hence my intention here was to be more specific.

> Are these two sentences referring to the same action? (i.e. is 'reconfiguring the Kommander controller...' referring to the same thing as 'Update the Kommander addon'?

No, reconfiguring kommander controller is part of the process of configuring kommander addon.
